packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts

```diff
@@ -372,8 +372,11 @@ const registryForNode = (node: Node): ShimmedCustomElementsRegistry | null => {
       scope = scopeForElement.get(scope)?.getRootNode() || document;
     }
   }
+  // @see https://github.com/webcomponents/polyfills/issues/624
+  // The problem is that NordVPN add "registry" property on the host element
+  // this can break if the host element has a "registry" property already but there is not solution for this
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  return (scope as any)['registry'] as ShimmedCustomElementsRegistry | null;
+  return ((scope as any)['registry'] || ((scope as any).host ? (scope as any).host['registry'] : null)) as ShimmedCustomElementsRegistry | null;
```

Comment:
Can you describe the problem and this solution a little more?

If NordVPN adds `registry` to the element (is that every element? Element.prototype? Some elements?), why would it interfere with `scope.registry`? Is it because NordVPN adds a particular kind of value to `registry`? What is it?

`scope` is usually a Document or ShadowRoot here. Is this only a problem when using innerHTML into a disconnected tree so that `scope` is an element? If so, could this be better handled in the if case above (line 372) where we know we have an Element and not a Document or ShadowRoot.

It's not obviously ok to me to be falling back to a ShadowRoot's host element to look for a registry, so I'm a little nervous about this change and want to understand it better.

Reply:
Yeah, I also don't understand either the problem or solution.
* #624 says the extension tampers with ShadowRoot's prototype but it doesn't say how exactly.
* the code here is falling back to looking for a registry on the host. That's not something the polyfill sets so where is that coming from? The extension?

We also need a test that sets up the DOM similar to the extension and demonstrates that this fix makes something work that otherwise wouldn't.